src/validators/dataclass.rs

```diff
@@ -571,13 +571,8 @@ impl DataclassValidator {
     ) -> ValResult<'data, PyObject> {
         // we need to set `self_instance` to None for nested validators as we don't want to operate on the self_instance
         // instance anymore
-        let val_output = state.with_new_extra(
-            Extra {
-                self_instance: None,
-                ..*state.extra()
-            },
-            |state| self.validator.validate(py, input, state),
-        )?;
+        let state = &mut state.rebind_extra(|extra| extra.self_instance = None);
```

Comment:
Is this changing anything semantically interesting?

Reply:
No, I just really hated `state.with_new_extra` so having found a better way here I used it where I can :)